tests/Integration/TelegramApiTest.php

```diff
@@ -543,7 +552,34 @@ public function check_the_webhook_works_and_can_emmit_an_event()
         //We can't pass test data to the webhook because it relies on the read only stream php://input
         $this->assertEmpty($update);
         $this->assertInstanceOf(Update::class, $update);
-        $emitter->emit(Argument::type(UpdateWasReceived::class))->shouldHaveBeenCalled();
+        $emitter->emit(Argument::type(UpdateWasReceived::class))->shouldHaveBeenCalledOnce();
+    }
+
+    /** @test */
+    public function it_emits_3_events_of_update_event_type()
+    {
+        $emitter = new Emitter();
+        $listener = $this->createSpyListener();
+        $emitter->addListener('*', $listener);
+
+        $api = $this->getApi();
+        $api->setEventEmitter($emitter);
+
+        $incomeWebhookRequest = $this->createIncomeWebhookRequestInstance([
+            'message' => [ // to help SDK to detect Update of "message" type and send 2nd event (with name "message")
+                'text' => 'any', // to help SDK to detect message type and send 3rd event (with name "message.text")
+            ],
+        ]);
+
+        $api->getWebhookUpdate(true, $incomeWebhookRequest);
+        $allEvents = $listener->events;
+
+        $this->assertArrayHasKey(UpdateEvent::NAME, $allEvents);
+        $this->assertArrayHasKey('message', $allEvents);
+        $this->assertArrayHasKey('message.text', $allEvents);
+        $this->assertCount(1, $allEvents[UpdateEvent::NAME]);
+        $this->assertCount(1, $allEvents['message']);
+        $this->assertCount(1, $allEvents['message.text']);
```

Comment:
3 events with same names as on SDK v4, event class has the same public methods (except it has `Api` instead `Bot` that is missing in SDK v3)